cpp/src/fil/infer.cu

```diff
@@ -792,8 +794,9 @@ template <int NITEMS,
           class storage_type>
 __global__ void infer_k(storage_type forest, predict_params params)
 {
+  using real_t = typename storage_type::real_t;
   extern __shared__ char smem[];
-  float* sdata       = (float*)smem;
+  real_t* sdata      = (real_t*)smem;
```

Comment:
Let's avoid C-style casts if we're already updating this line.

Reply:
Done.